### ISSUE TITLE
Fix KNC specific magic numbers

### ISSUE THREAD

**Issue by bjoo** [JeffersonLab/qphix#19]
As we evolve the specific KNC magic numbers of 240 threads may become inappropriate, even for KNC where we can have e.g. 244 in some cases. I should fix this, especially in the BLAS/Inner Product section where it can cause mayhem.

**Comment by martin-ueding** [comment]
What about things like `_mm_prefetch((const char *)( ((*clBase).diag1[ 0 ] +240) ), _MM_HINT_T0);` in the AVX512 kernels? Is that `240` also a KNC magic number?

**Comment by ddkalamk** [comment]
No, this is simply offset for prefetching from base address and nothing to do with KNC. Should be same for all ISAs.
